# data set Titanic


## importamos las librerias necesarias para el data set 

In [101]:
import pandas as pd

## analisis inicial 

### leemos el archivo csv train.csv


In [102]:
df = pd.read_csv('train.csv')

### mostramos las dimensiones del data set


In [103]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


### df.info()

despues de ejecutar el df.info() obtuvimos la siguiente información 
tenemos 12 columnas las cuales estan dividas en: id del pasajero, sobrevivido, clase en la que estaba (clase siendo referencia a la clase social en la que se estaba alojando el tripulante), que edad tenia, si viajaba con un acompañante o no, el numero de ticket, el puerto de embarque, tarifa, sexo, nombre, cuantos padres o hijos viajaban en la embarcacion.

### ¿cuantos datos nulos tenemos?

In [104]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

en las casillas donde se muestran datos vacios son las siguientes: 
1. Age con 177 datos nulos 
2. cabin con 687 datos nulos
3. embarked con 2 datos nulos

### df.head

con df.head() nos muestra los primeros 5 registros del data set 

In [105]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

## estrategia para la columna Age (edad)

se utilizara la mediana por que es la que permite encontrar el valor en el centro y para los datos que tenemos es la mas optima 

In [106]:
# Seleccionas la columna 'Age' y luego calculas su mediana
mediana_edad = df['Age'].median()

print(f"La mediana de la edad es: {mediana_edad} años ")

""" # Es un atajo para df['Age'].median()
mediana_edad = df.Age.median()

print(f"La mediana de la edad es: {mediana_edad}")"""

La mediana de la edad es: 28.0 años 


' # Es un atajo para df[\'Age\'].median()\nmediana_edad = df.Age.median()\n\nprint(f"La mediana de la edad es: {mediana_edad}")'

#### rellenar los valores faltantes



In [107]:
df['Age'].fillna(value=mediana_edad, inplace = True)

C:\Users\PC\AppData\Local\Temp\ipykernel_14476\1939184208.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Age'].fillna(value=mediana_edad, inplace = True)


In [108]:
df['Age'].isnull().sum()

np.int64(0)